File: Challenges/Coding World Cup/2023/Day 6 - ENG vs BAN - Jersey Swap.cpp
int replaceElements(vector<int> &a)
{
    // Write your code here
    set<int> set;
    for (int i = 0; i < a.size(); i++)
    {
        if (set.find(a[i]) != set.end())
        {
            set.erase(a[i]);
        }
        else
        {
            set.insert(a[i]);
        }
    }

    if (set.size() % 2 == 0)
        return set.size() / 2;
    return -1;
}

File: Challenges/Coding World Cup/2023/Day 6 - ENG vs BAN - Jersey Swap.java
import java.util.*;

public class Solution {
    static int replaceElements(int[] a) {
        // Write your code here.
        HashMap<Integer, Integer> hm = new HashMap<>();
        int n = a.length;

        if (n % 2 != 0) {
            return -1;
        }

        int count = 0;
        for (int i = 0; i < n; i++) {
            if (hm.containsKey(a[i])) {
                hm.put(a[i], hm.get(a[i]) + 1);
            } else {
                hm.put(a[i], 1);
            }
        }

        for (Map.Entry<Integer, Integer> entry : hm.entrySet()) {
            if (entry.getValue() % 2 != 0) {
                count++;
            }
        }

        return count % 2 == 0 ? count / 2 : count / 2 + 1;
    }
}